File: Core/actor/digitalactor.cpp
#include "digitalactor.h"

#include <QDebug>

DigitalActor::DigitalActor() : ActorBase() {
}

DigitalActor::DigitalActor(ValueGroup *valueGroup, QString id, VALUE_TYPE valueType, bool isAsync, QObject *parent) : ActorBase(valueGroup, id, valueType, QVariant::Bool, parent), m_isAsync(isAsync)
{

}


bool DigitalActor::cmdSupported(actor::ACTOR_CMDS cmd) {
    switch(cmd) {
    case actor::ACTOR_CMD_ON:
    case actor::ACTOR_CMD_OFF:
        return true;
    default:
        return false;
    }
}

QVariant DigitalActor::_updateValue(QVariant newValue) {
    iDebug() << Q_FUNC_INFO << newValue;

    if (newValue.canConvert(QVariant::Bool)) {
        return QVariant::fromValue(newValue);
    } else {
        return QVariant();
    }
}

bool DigitalActor::isAsync() {
    return m_isAsync;
}

void DigitalActor::_triggerCmd(actor::ACTOR_CMDS cmd) {
    Q_EMIT(statusRequested(cmd == actor::ACTOR_CMD_ON));

    if (!m_isAsync) {
        updateValue(cmd == actor::ACTOR_CMD_ON);
    }
}

File: AndroidApp/OshCore/src/main/java/com/osh/actor/DigitalActor.java
package com.osh.actor;

import com.osh.SerializationSupport;
import com.osh.value.ValueGroup;
import com.osh.value.ValueType;

public class DigitalActor extends ActorBase implements SerializationSupport {
	
	protected boolean isAsync;

	public DigitalActor(ValueGroup valueGroup, String id, ValueType valueType, boolean isAsync) {
		super(valueGroup, id, valueType);
		this.isAsync = isAsync;
	}

	public DigitalActor() {
		super();
	}

	@Override
	public boolean cmdSupported(ActorCmds cmd) {
	    switch(cmd) {
	    case ACTOR_CMD_ON:
	    case ACTOR_CMD_OFF:
	        return true;
	    default:
	        return false;
	    }
	}

	@Override
	protected void _triggerCmd(ActorCmds cmd) {
	    switch(cmd) {
	    case ACTOR_CMD_ON:
	        //Q_EMIT(statusRequested(true));
	        break;
	    case ACTOR_CMD_OFF:
	        //Q_EMIT(statusRequested(false));o
	        break;
	    default:
	        break;
	    }
	}
	
	protected Object _updateValue(Object newValue) {
	    if (newValue instanceof Boolean) {
	        return newValue;
	    } else {
	        return null;
	    }
	}

}
